src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey
+<T>(items: T[], key: keyof T): GroupsMap<T> {
+  // i know it is not good, but what type it will be?
+  const result: { [key: string]: T[keyof T] } = {};
+
+  for (let i: number = 0; i < items.length; i += 1) {
+    if (typeof result[items[i][key]] === 'undefined') {
+      result[items[i][key]] = [];
+    }
+
```

Comment:
Better will be Array.forEach 
No need to check on typeof 
`if (!result[items[i][key]])`